### DISCUSSION TITLE
Unspecific enzyme searches

### DISCUSSION THREAD

**Discussion by tharan-srikumar-bruker** [wfondrie/mokapot#90] [Q&A]
A rather naïve question: how much effort would it be to support unspecific enzyme searches, such as those used in immunopeptidomics experiments?  Is it a matter of parameter tweaks or something larger?  Any plans to support this type of workflow?

**Comment by wfondrie** [comment]
Hi @tharan-srikumar-bruker 👋 

At the PSM- and peptide-levels, there should be no difference for mokapot between a unspecific enzyme search and those using an enzyme like trypsin. However, the protein-level is a little more complicated.

Mokapot uses the "[picked-protein approach](https://pubmed.ncbi.nlm.nih.gov/25987413/)" to perform protein-level confidence estimation. Part of this process involves determining the unique and shared possible peptides for every protein the FASTA file, not only those that were detected. Proteins with proper subsets of peptides are grouped and only unique peptides are used for the remaining. There are two complications this can cause in the case of a nonspecific enzyme search:
- Many of the non-enzymatic peptides are likely to be shared between more than one protein, which means it will not be used to support the detection of either protein.
- From a computational standpoint, mokapot will store all of the peptides in memory. This may be slow and you'll need to run it a on a machine with plenty of memory (I don't know exactly how much off the top of my head).

That being said, these concerns are merely theoretical:  I've never actually attempted running mokapot on a non-specific enzyme dataset, so I'm curious how it will perform! To actually run mokapot on a non-specific enzyme dataset from the command line run:

```
mokapot --enzyme '.'  psms.pin 
```
Also make sure to match the other digestion parameters to those used during your search. My advice would be, if you try it, use a FASTA file that includes only canonical sequences, not isoforms. Also, please let me know how it goes!

**Reply by tharan-srikumar-bruker** [reply]
Just wanted to report back that as you indicated we ran into memory issues.  We were running on a box with 128GB and crashed out with too many hard faults.  Will revisit at a later time.
